gradoop-flink/src/main/java/org/gradoop/flink/model/impl/operators/sampling/RandomNonUniformVertexSampling.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright © 2014 - 2018 Leipzig University (Database Research Group)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradoop.flink.model.impl.operators.sampling;
+
+import org.apache.flink.api.java.DataSet;
+import org.gradoop.common.model.impl.pojo.Edge;
+import org.gradoop.common.model.impl.pojo.Vertex;
+import org.gradoop.flink.algorithms.gelly.vertexdegrees.DistinctVertexDegrees;
+import org.gradoop.flink.model.api.epgm.LogicalGraph;
+import org.gradoop.flink.model.api.operators.UnaryGraphToGraphOperator;
+import org.gradoop.flink.model.impl.functions.epgm.Id;
+import org.gradoop.flink.model.impl.functions.epgm.SourceId;
+import org.gradoop.flink.model.impl.functions.epgm.TargetId;
+import org.gradoop.flink.model.impl.functions.utils.LeftSide;
+import org.gradoop.flink.model.impl.operators.sampling.functions.AddMaxDegreeCrossFunction;
+import org.gradoop.flink.model.impl.operators.sampling.functions.NonUniformVertexRandomFilter;
+import org.gradoop.flink.model.impl.operators.sampling.functions.RemoveUnnecessaryPropertiesMap;
+import org.gradoop.flink.model.impl.operators.sampling.functions.VertexToDegreeMap;
+
+import java.util.Vector;
+
+/**
+ * Computes a vertex sampling of the graph. Retains randomly chosen vertices of a given relative
+ * amount and all edges which source- and target-vertices where chosen. A degree-dependent value
+ * is taken into account to have a bias towards high-degree vertices. There may retain some
+ * unconnected vertices in the sampled graph.
+ */
+public class RandomNonUniformVertexSampling implements UnaryGraphToGraphOperator {
+  /**
+   * relative amount of vertices in the result graph
+   */
+  private final float sampleSize;
+
+  /**
+   * seed for the random number generator
```

Comment:
Sometimes start your comment in upper case sometimes not. Pls make that consistent.

Reply:
I changed all to upper case.